### ISSUE TITLE
About Training Settings

### ISSUE THREAD

**Issue by Huanxi2018** [Kovelja009/mixed_precision_dnn_pytorch#1]
Thanks for your job! 
When I train the resnet-18 using ''python training.py PARAMETRIC_FP_WAQ_DELTA_XMAX_RLR --cfg train_resnet_quant_fp.cfg'' that you have provided, the training acc just holds up 10% after the 10 epochs and 8 bit is all it learned.
![image](https://github.com/Kovelja009/mixed_precision_dnn_pytorch/assets/49297333/9bcb7770-449c-4bec-a5ca-1885f90302db)

I wonder why the epoch is set to 10 without learning rate dropping scheduler. By the way, if I want to change cifar-10 to imagenet, could I just replace the dataset settings and add the learning rate dropping scheduler without anything else to change? 
Looking forward to your reply.

**Comment by Huanxi2018** [comment]
> Thanks for your job! When I train the resnet-18 using ''python training.py PARAMETRIC_FP_WAQ_DELTA_XMAX_RLR --cfg train_resnet_quant_fp.cfg'' that you have provided, the training acc just holds up 10% after the 10 epochs and 8 bit is all it learned. ![image](https://private-user-images.githubusercontent.com/49297333/311482823-9bcb7770-449c-4bec-a5ca-1885f90302db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAwNTQyODcsIm5iZiI6MTcxMDA1Mzk4NywicGF0aCI6Ii80OTI5NzMzMy8zMTE0ODI4MjMtOWJjYjc3NzAtNDQ5Yy00YmVjLWE1Y2EtMTg4NWY5MDMwMmRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEwVDA2NTk0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiODEyYmZkNjI4MTAyODY1YzRkNDVmMjc3OTVkYmM1YzIxYmQ4MTc2Yzc5ZmE0YjEzYjIzYTJiMmVmYmE5NzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Uyzyf1S0RItVBMDpHnkczX0CmmzL7Wf3jzYI6QT526c)
> 
> I wonder why the epoch is set to 10 without learning rate dropping scheduler. By the way, if I want to change cifar-10 to imagenet, could I just replace the dataset settings and add the learning rate dropping scheduler without anything else to change? Looking forward to your reply.

**Comment by Kovelja009** [comment]
Hi, this is still work in progress, so I have a bug where during backpropagation, bitwidths for different layers are not updated for some reason and I should fix it in the future. This is still in testing stage so I have hardcoded 10 epochs and dataset just to see whether bitwidths are being updated. Mixed-precision learning is written as a PyTorch module, so ideally it should work on any network and you can change learning rate scheduler, network and dataset. Hope this answered your question and feel free to ask if there is anything else you are interested in :)

**Comment by Huanxi2018** [comment]
> Hi, this is still work in progress, so I have a bug where during backpropagation, bitwidths for different layers are not updated for some reason and I should fix it in the future. This is still in testing stage so I have hardcoded 10 epochs and dataset just to see whether bitwidths are being updated. Mixed-precision learning is written as a PyTorch module, so ideally it should work on any network and you can change learning rate scheduler, network and dataset. Hope this answered your question and feel free to ask if there is anything else you are interested in :)

Thank you very much for your quick reply. I will keep an eye on your work~
